### PR TITLE
[Draft] Adopt a threading model for smoke tests.

### PR DESCRIPTION
**Pull request by JoshLind**
## Motivation

(Write your motivation for proposed changes here.)

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

## Test Plan

(Share your test plan here. If you changed code, please provide us with clear instructions for verifying that your changes work.)

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/libra/libra/developers.libra.org, and link to your PR here.)

### CODE REVIEW

File: testsuite/smoke-test/src/workspace_builder.rs [hunk 1]
```diff
@@ -95,37 +95,3 @@ pub fn get_bin<S: AsRef<str>>(bin_name: S) -> PathBuf {
...
-    let bin_path = build_dir().join(format!("{}{}", "libra-node", env::consts::EXE_SUFFIX));
-    if !bin_path.exists() {
-        panic!(format!(
-            "Can't find binary libra-node in expected path {:?}",
-            bin_path
-        ));
-    }
-
-    bin_path
-}
```
Comment (davidiw) [inline testsuite/smoke-test/src/workspace_builder.rs:131]: let's do this, please

**Review by davidiw — COMMENTED** [review]
is there a blocker here?

**Comment by davidiw** [comment]
let's close this for now

**Comment by JoshLind** [comment]
@davidiw, agreed. I think there's too many things blocking us from deploying this well enough. Perhaps I'll revisit it later at some point 😄. Thanks!

**Pull request closed without merging**